opendbc/car/tesla/carstate.py

```diff
@@ -0,0 +1,112 @@
+import copy
+from opendbc.can.can_define import CANDefine
+from opendbc.can.parser import CANParser
+from opendbc.car import structs
+from opendbc.car.common.conversions import Conversions as CV
+from opendbc.car.interfaces import CarStateBase
+from opendbc.car.tesla.values import DBC, CANBUS, GEAR_MAP
+
+ButtonType = structs.CarState.ButtonEvent.Type
+
+class CarState(CarStateBase):
+  def __init__(self, CP):
+    super().__init__(CP)
+    self.can_define = CANDefine(DBC[CP.carFingerprint]['pt'])
+
+    self.hands_on_level = 0
+    self.steer_warning = None
+    self.das_control = None
+
+  def update(self, cp, cp_cam, *_) -> structs.CarState:
+    ret = structs.CarState()
+
+    # Vehicle speed
+    ret.vEgoRaw = cp.vl["ESP_B"]["ESP_vehicleSpeed"] * CV.KPH_TO_MS
```

Comment:
This is only 0.5 kph resolution, do we have anything else?

![image](https://github.com/user-attachments/assets/01993f63-d656-47f3-a353-49baf6c2400d)

Reply:
This one looks much better 
![image](https://github.com/user-attachments/assets/5971bfbc-7bf0-485b-b08d-31f542f6cc1e)

Reply:
I will also change it later in the panda code too